algosdk/transaction.py

```diff
@@ -368,9 +368,8 @@ def __init__(
             raise error.WrongAmountType
         self.close_remainder_to = close_remainder_to
         if not sp.flat_fee:
-            self.fee = max(
-                self.estimate_size() * self.fee, constants.min_txn_fee
-            )
+            mf = sp.min_fee or constants.min_txn_fee
```

Comment:
I believe `constants.min_txn_fee` will be used if `sp.min_fee` is 0? I'd rather only fall back to the constant if `sp.min_fee` is None

Reply:
I believe flat fee in both of these SDKs is a boolean which indicates if fee should be taken as-is (even if it's below min fee) or interpreted as a fee-per-byte.

So there are two routes to creating a 0 fee transaction
1. Specify `fee=0` and `min_fee=0`
2. Or, specify `fee=0` and `flat_fee=True` (`min_fee` doesn't matter in this case)

>  I worried people might make SuggestedParams objects with 0 when they meant "I don't care about that"

IMO omitting it completely or setting it to None is more indicative of not caring than setting it to 0 explicitly. In a language like python where any value can be None, I'd rather use that as a special case than 0